rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -2740,6 +2739,10 @@ int LuaUnsyncedCtrl::WarpMouse(lua_State* L)
  */
 int LuaUnsyncedCtrl::SetMouseCursor(lua_State* L)
 {
+	// ignore lua requests to control cursor when mouse is over RmlUI element
+	if (RmlGui::IsActive())
```

Comment:
If it's already possible in Lua that one widget changes cursor when mouse is over a different one, then this change is a breaking behavior.

Reply:
The widget that's caused me trouble- that made me include that line is `cmd_rclick_quick_build_resource_extractor.lua` because it sets the cursor when you're hovering over a metal spot

Looking into things more- it seems that the last widget to set the cursor "wins".

All the gui widgets have logic along the lines of  "if the mouse is over me, set the cursor to 'cursornormal' every frame" in either `widget:Update()` or `widget:DrawScreen()`

RmlUI only calls SetMouseCursor when it determines internally that there's been change in shown cursor caused by hovering over something. That results in the cursor changing for one frame before a widget sets it to something else.

It does **not** act like the widgets which fight over the cursor icon every frame.
That check makes the widgets fighting over the mouse cursor get completely ignored.

Reply:
Also, now that I think about, `IsActive()` is a really bad name now for what that check does.

I'll rename that to `IsMouseAbove()` or some such

Reply:
I could also do a thing where I track the last requested cursor from RmlUi

This works because RmlUI sets the cursor to an empty string as the last-case scenario

https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/contexts.html#mouse-cursor

that way an Rml element could just not do anything to the cursor icon- letting Widgets and other things have control

Reply:
I've used that implementation detail to refactor how the mouse cursor eventually gets set by RmlUi
If Rml is being interacted with and it wants a certain cursor- the CMouseHandler::Update() code will respect that.